apps/websockets/viewsets.py

```diff
@@ -0,0 +1,160 @@
+        chat_object: Chat = self.get_object()
+        messages = Message.objects.filter(chat_id=chat_object.id)
+        page = self.paginate_queryset(messages)
+        serializer = self.get_serializer(page, many=True)
+        return self.get_paginated_response(serializer.data)
+
+    @swagger_auto_schema(
+        request_body=no_body
+    )
+    @action(
+        methods=['get'],
+        detail=True,
+        url_path='participants',
+        url_name='chat_participants'
+    )
+    def participants(self, request, chat_id):
+        chat_object: Chat = self.get_object()
+        participants = user_model.objects.filter(
+            id__in=chat_object.participants.all().values_list("id", flat=True)
+        )
+        page = self.paginate_queryset(participants)
+        serializer = self.get_serializer(page, many=True)
+        return self.get_paginated_response(serializer.data)
+
+    @action(
+        methods=['post'],
+        detail=True,
+        url_path='send_message',
+        url_name='send_message',
+    )
+    def send_message(self, request, chat_id):
+        """
+        CREATE method for messages but with sending it
+        via websockets to the provided chat if exists
+        """
+        chat_object = self.get_object()
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        message_object = Message.objects.create(
+            created_by=request.user,
+            chat=chat_object,
+            message_text=serializer.validated_data.get("message_text")
+        )
+
+        ChatManager.chat_message(message_object)
+
+        return Response(
+            status=status.HTTP_201_CREATED,
+            data=model_to_dict(message_object),
+        )
+
+    @swagger_auto_schema(auto_schema=None)
+    def create(self, request, *args, **kwargs):
+        pass
+
+
+class MessageViewSet(viewsets.ModelViewSet):
+    model = Message
+    permission_classes = [IsAuthenticated, MessagePermissions]
+    lookup_url_kwarg = "message_id"
+    http_method_names = ["get", "patch", "delete"]
+    queryset = model.objects.all()
+
+    def get_serializer_class(self):
+        match self.action:
+            case "retrieve":
+                return MessageRetrieveSerializer
+            case "list":
+                return MessageRetrieveSerializer
+            case "partial_update":
+                return MessageCreateSerializer
+            case _:
+                return EmptySerializer
+
+    def get_queryset(self):
+        if self.kwargs.get("message_id"):
+            self.queryset = self.model.objects.all()
+        else:
+            self.queryset = self.model.objects.filter(
+                Q(created_by=self.request.user) |
+                Q(id__in=Chat.objects.filter(
+                    participants__id=self.request.user.id
+                ).values_list("chat_messages", flat=True))
+            )
+        return self.queryset.all()
```

Comment:
We also need a notifications viewset for the notification bell task

Reply:
There are two different ways:
- I'll do this task as well
- I'll end the notifications polishing along with the chat task but the viewset creation could take another person (Evgeniy for example, he is interested in websockets)

Reply:
It would be nice just to give a frontend ability to get list of notifications for specific user, if you are able to add this during this task, it would greatly improve the scope of work we can delegate to frontend devs